compiler/rustc_hir/src/hir.rs

```diff
@@ -417,6 +422,14 @@ impl<'hir> GenericArgs<'hir> {
     }
 }
 
+#[derive(Copy, Clone, PartialEq, Eq, Encodable, Hash, Debug)]
+#[derive(HashStable_Generic)]
+pub enum GenericArgsParentheses {
+    No,
+    ReturnTypeNotation,
+    ParenSugar,
```

Comment:
It might be worth a doc comment on at least `ReturnTypeNotation` and `ParenSugar` to show what these correspond to in the surface Rust syntax.